File: src/migrations/migration.ts
import {Kingdom} from '../kingdom/data/kingdom';
import {Camping} from '../camping/camping';

export abstract class Migration {
    protected constructor(public readonly version: number) {
    }

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    async migrateKingdom(game: Game, kingdom: Kingdom): Promise<void> {

    }

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    async migrateCamping(game: Game, camping: Camping): Promise<void> {

    }

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    async migrateOther(game: Game): Promise<void> {

    }
}

File: src/jsMain/kotlin/at/posselt/pfrpg2e/migrations/migrations/Migration.kt
package at.posselt.pfrpg2e.migrations.migrations

import com.foundryvtt.core.Game

abstract class Migration(
    val version: Int,
) {
    open suspend fun migrateCamping(game: Game, camping: dynamic) {

    }

    open suspend fun migrateKingdom(game: Game, kingdom: dynamic) {

    }

    open suspend fun migrateOther(game: Game) {

    }
}